### PR TITLE
Actionized classes can be procs

### PR DESCRIPTION
**Pull request by MissingHandle**
I got to thinking about this last night.  Makes Actionizer Even.More.Functional.

### CODE REVIEW

**Comment by mikenichols** [comment]
The tests are failing due to rubocop style issues. They're all pretty simple though.

**Comment by MissingHandle** [comment]
@mikenichols - note that this breaks compatibility with ruby 2.0

**Comment by mikenichols** [comment]
@MissingHandle Do you propose we require ruby 2.2 now? https://www.sitepoint.com/new-methods-ruby-2-2/ for reference

**Comment by mikenichols** [comment]
I threw a `binding.pry` between lines 54 and 55 of `lib/actionizer.rb` to check that the inputs are passed correctly. `input.to_h` gives me an empty hash. Would you need to do something like `[{ named_arg: 1}].map { |e| class_with_call.call(e) }` to get the arguments passed in? In that case, what's the benefit of adding this `to_proc` method?

**Comment by MissingHandle** [comment]
daammit - I didn't test that it could receive args :(.  okay, will come back to this / figure out if we can do it or not.

File: lib/actionizer.rb [hunk 1]
```diff
@@ -12,6 +12,14 @@ def self.included(base)
   end
 
   module ClassMethods
+    def to_proc
+      if new.respond_to?(:call)
+        method(:call).curry
```
Comment (MissingHandle) [inline lib/actionizer.rb]: i think we just need to curry it with the inputs hash here.

Comment (mikenichols) [inline lib/actionizer.rb]: We don't have `input` here. It's only created when you go through `method_missing` with args. So whenever you say `collection.map(&my_action_class)`, you're not passing args and so there'll be no `input`.

Comment (MissingHandle) [inline lib/actionizer.rb]: I think... 
```
klass = self
Proc.new { |caller| klass.new(caller).call }
```

**Pull request closed without merging**